src/patient/PatientInfoFields.tsx

```diff
@@ -0,0 +1,84 @@
+import { Field, ErrorMessage, FormikState } from 'formik';
+import { differenceInYears } from 'date-fns';
+
+export interface PatientInfo {
+  name: string;
+  dateOfBirth: string;
+  eye: 'left' | 'right';
+}
+
+const calcAge = (dateOfBirth: string | undefined, error: string | undefined) =>
+  dateOfBirth && !error
+    ? differenceInYears(new Date(), new Date(dateOfBirth)) || ''
+    : '';
+
+export const PatientInfoFields = ({
+  errors,
+  values,
+  touched
+}: FormikState<PatientInfo | undefined>) => (
+  <>
+    <h2>Patient</h2>
+    <div className="form-row">
+      <div className="form-group col-md-4">
+        <label htmlFor="fieldName">Name</label>
+        <Field
+          name="patient.name"
```

Comment:
This `patient.name` reference is ugly, because the component needs to be aware of the ancestor structure. Is there a more elegant way of doing this?

Reply:
https://formik.org/docs/guides/arrays